src/App.tsx

```diff
@@ -16,35 +22,101 @@ export const goodsFromServer = [
 ];
 
 export const App: React.FC = () => {
+  const [goods, setGoods] = useState<string[]>(goodsFromServer);
+  const [selectedSort, setSelectedSort] = useState<SortType | null>(null);
+  const [isReverse, setIsReverse] = useState<boolean>(false);
+
+  function sorting(sortingType: SortType | null): void {
+    let sortedGoods: string[] = [...goodsFromServer];
+
+    switch (sortingType) {
+      case SortType.SORT_BY_ALPHABET:
+        sortedGoods = [...goods].sort((good1, good2) =>
+          good1.localeCompare(good2),
+        );
+        break;
+      case SortType.SORT_BY_LENGTH:
+        sortedGoods.sort((good1, good2) => {
+          if (good1.length === good2.length) {
+            return good1.localeCompare(good2);
+          }
+
+          return good1.length - good2.length;
+        });
+        break;
+      default:
+        return;
+    }
+
+    if (isReverse) {
+      sortedGoods.reverse();
+    }
+
+    setGoods(sortedGoods);
+    setSelectedSort(sortingType);
+  }
```

Comment:
The function's name `sorting` does not start with a verb that clearly indicates what the method does. Consider renaming it to `sortGoods` to better reflect its action, in accordance with the [CODE STYLE] checklist item.